tsconfig.json

```diff
@@ -1,17 +1,12 @@
 {
   "compilerOptions": {
-    "lib": ["es6"],
+    "lib": ["dom", "es2017"],
```

Comment:
Some questions here:
1 - Why incldue DOM here but not in tsconfig.release.json?
2 - This ends up exlcuding a few things that es6 would include - per https://www.typescriptlang.org/docs/handbook/compiler-options.html it looks like 'es6' includes DOM.iterable and ScriptHost and this does not. If we don't use these libraries, I like excluding these since it should help reduce the file size for deployments. Are we sure these are unused?

Reply:
> 1 - Why incldue DOM here but not in tsconfig.release.json?

I had to add it as `protobufjs` shipped without these definitions, see https://travis-ci.org/firebase/firebase-functions/jobs/536773800. I just tested it with the newest version and it's fixed, so I removed it: 8fd948b6828865a1d88e5a3b40ba0db01a8ccff3

> 2 - This ends up exlcuding a few things that es6 would include - per https://www.typescriptlang.org/docs/handbook/compiler-options.html it looks like 'es6' includes DOM.iterable and ScriptHost and this does not. If we don't use these libraries, I like excluding these since it should help reduce the file size for deployments. Are we sure these are unused?

This target was repeatedly specified by core TypeScript maintainers as the correct one for Node.js 8, there's also a PR open by one of them: https://github.com/microsoft/TypeScript-wiki/pull/182

`lib` specifies available functions and properties, it does not affect transpilation in any way, so if it compiles (and the CI is green) we are good.

Reply:
I'm fine with `es2017` which I see you've changed this to. Resolving.